### PR TITLE
install apt-transport-https via required_packages (issue #369)

### PR DESCRIPTION
**Pull request by madddi**
Installing apt-transport-https via required_packages mitigates the cyclic dependency mentioned in issue #369

### CODE REVIEW

**Comment by jkroepke** [comment]
required_packages is in apt module deprected and emitts a warning.

File: manifests/repos.pp [hunk 1]
```diff
@@ -7,26 +7,24 @@
...
-        apt::source { 'docker':
-          location          => $docker::package_source_location,
-          release           => $docker::package_release,
-          repos             => $docker::package_repos,
-          key               => $docker::package_key,
-          key_source        => $docker::package_key_source,
-          required_packages => 'debian-keyring debian-archive-keyring',
-          pin               => '10',
-          include_src       => false,
+      if $docker::manage_package {
```
Comment (fraenki) [inline manifests/repos.pp]: +1 Finally this would enable us to manage apt on our own.

**Comment by garethr** [comment]
@madddi thanks for looking at this. I'm having trouble reviewing as it looks like the commit history contains lots of commits from master. If you could rebase against the current master I'll take a closer look and provide feedback. Thanks again.

**Comment by madddi** [comment]
@garethr Sorry, I unfortunately did'nt get around to update this PR. As @jkroepke pointed out, "required_packages" is deprecated in the apt module. Looking at this again I think the best solution would be to provide "manage_repo" and "manage_package" options to disable repository management separately. If you're ok with this I'd like to close this request and open another one with the changes described above.

**Comment by garethr** [comment]
@madddi I think that sounds like a good idea. I'll close now as mentioned. And I'll look out for the PR and try and jump on a review quickly. Thanks

**Pull request closed without merging**